hypothesis/lms
# Refactor the `VitalSource` service to split out a client

For:

 * https://github.com/hypothesis/lms/issues/4214

There's a lot going on here, but basically:

* Hide away any methods which aren't public
* Split the client and the service so that:
   * The service is the public interface
   * The client makes individual calls to the API
* Improvements around the HTTPService to allow it to be used as a session
   * This allows setting headers once instead of on every call
* Split out a small helper for dealing with VS book locations (this untangled some mess)

### About the size

 * This has gotten a little bigger as a result of duplicated imports and testing interfaces that weren't there before
 * We've also added a set of proxy methods in the service
 * We've also added some doc strings
 * Overall it's not much bigger because some of the refactors reduced the size

## Testing notes

 * Get everything bang up to date and started
 * Make sure you have run `make devdata`
 * Visit: https://hypothesis.instructure.com/courses/125/assignments and create a new localhost assignment
 * Choose Vitalsource and add https://hypothesis.vitalsource.com/reader/books/9781786894700
 * You should see the cover and the name
 * Continue on and you should be able to pick a chapter
 * Save and publish and you should see the book
 * You **will not** see the client as it's not available in local dev yet
